libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -123,15 +147,13 @@ IntentFilter obtainEventReceiverIntentFilter() {
     return eventReceiverIntentFilter;
   }
 
-  private void initializeTelemetryClient(String accessToken, String userAgent) {
-    if (isTelemetryClientInitialized()) {
-      telemetryClient = createTelemetryClient(accessToken, userAgent);
-      queue.setTelemetryInitialized(true);
-    }
+  boolean checkRequiredParameters(String accessToken, String userAgent) {
+    return isAccessTokenValid(accessToken) && isUserAgentValid(userAgent);
   }
 
-  private boolean isTelemetryClientInitialized() {
-    return accessToken != null && !accessToken.isEmpty();
+  private void initializeTelemetryClient() {
```

Comment:
We should revisit this because we could take advantage of the `checkRequiredParameters` check to not create the telemetry client and only do that if both the `accessToken` and the `userAgent` are valid so we don’t assign memory if not needed but carefully because we should still be calling `queue.setTelemetryInitialized(true)` in order to keep queuing events although we don’t send them.